### DISCUSSION TITLE
Annotation tool color picker not sampling in linear space when using OCIO

### DISCUSSION THREAD

**Discussion by markreidvfx** [AcademySoftwareFoundation/OpenRV#407] [Ideas]
Currently the annotation tool's color picker samples pixel values after the display transform has been applied with [glReadPixels](https://github.com/AcademySoftwareFoundation/OpenRV/blob/f3066e75227aecc632412cec1b3fb5ac03cdc65e/src/plugins/rv-packages/annotate/annotate_mode.mu#L676).  When you draw and are using OCIO, the color you sample will appear wrong because the `OCIODisplay` gets applied a second time.  This is because the RV paint node is placed between the `OCIOFile` and  `OCIODisplay` transforms in the node graph. 

![pipeline](https://github.com/AcademySoftwareFoundation/OpenRV/assets/814966/bf5ead66-3f74-4cdc-a042-183f490352ae)

https://github.com/AcademySoftwareFoundation/OpenRV/assets/814966/db990a7e-399d-4684-bb9f-6cbd266d3606

Is there a way or can we add a way to sample a pixel's linear value, preferably after the paint node. How about being able to do it at any arbitrary place in the node graph too? I assume it hasn't been done because everything is in the gpu at this point and it would require rendering/sync to cpu visible buffer.

I've worked around this internally by using `sourcePixelValue` to get the raw pixel value. Then I applied the same `OCIOFIle` transform via OCIO's python api. This unfortunately doesn't let you sample from previously painted brush strokes. I consider reversing the display lut but then I'd also have to consider the other effects (like exposure, saturation) applied to the linear image.

**Comment by mcoliver** [comment]
Hey @markreidvfx  A quick clarification, do you want to expose this functionality under the hood, or do you want this to be user selectable in the UI?  Or do you want the color picker to return a JSON of all the possible colors at all the layers? Or something else?
